### PR TITLE
feature/491

### LINKED ISSUE #491: buildWaypointModelsForProcedure() builds from sidCollection only when flightPhase is APRON
```javascript
buildWaypointModelsForProcedure(procedureRouteSegment, runway, flightPhase) {
    const routeModel = new RouteModel(procedureRouteSegment);
    let standardRouteWaypointModelList;

    // TODO: As amended, this may be an unsafe assumption. Needs to be reexaimed.
    if (flightPhase === FLIGHT_PHASE.APRON) {
        standardRouteWaypointModelList = this._sidCollection.generateFmsWaypointModelsForRoute(
            routeModel.procedure,
            runway,
            routeModel.exit
        );
    } else {
        standardRouteWaypointModelList = this._starCollection.generateFmsWaypointModelsForRoute(
            routeModel.procedure,
            routeModel.entry,
            runway
        );
    }

    return standardRouteWaypointModelList;
}
```

### PR DESCRIPTION
**Pull request by n8rzz**
Resolves #491.

Adds .isAirborneFlightPhase() and implements this helper in .buildWaypointModelsForProcedure()

### CODE REVIEW

File: src/assets/scripts/client/navigationLibrary/NavigationLibrary.js [hunk 1]
```diff
@@ -233,7 +233,7 @@ export default class NavigationLibrary {
         let standardRouteWaypointModelList;
 
         // TODO: As amended, this may be an unsafe assumption. Needs to be reexaimed.
-        if (flightPhase === FLIGHT_PHASE.APRON) {
+        if (this.isAirborneFlightPhase(flightPhase)) {
```
Comment (erikquinn) [inline src/assets/scripts/client/navigationLibrary/NavigationLibrary.js]: This makes me slightly nervous limiting it to conditions like these...

Also, don't you want the inverse condition here?

File: src/assets/scripts/client/navigationLibrary/NavigationLibrary.js [hunk 2]
```diff
@@ -302,4 +302,19 @@ export default class NavigationLibrary {
...
+     *
+     * @for NavigationLibrary
+     * @method isAirborneFlightPhase
+     * @param flightPhase {string}
+     * @return {boolean}
+     */
+    isAirborneFlightPhase(flightPhase) {
+        return flightPhase === FLIGHT_PHASE.APRON ||
+            flightPhase === FLIGHT_PHASE.TAXI ||
+            flightPhase === FLIGHT_PHASE.WAITING;
```
Comment (erikquinn) [inline src/assets/scripts/client/navigationLibrary/NavigationLibrary.js]: These should be the opposite, or the function renamed appropriately.

File: test/navigationLibrary/NavigationLibrary.spec.js [hunk 3]
```diff
@@ -54,3 +54,21 @@ ava('.isSuffixRoute() returns true when passed a routeString associated with a s
...
+    t.true(navigationLibrary.isAirborneFlightPhase('APRON'));
+});
+
+ava('.isAirborneFlightPhase() returns true when passed `TAXI`', (t) => {
+    const navigationLibrary = new NavigationLibrary(AIRPORT_JSON_KLAS_MOCK);
+
+    t.true(navigationLibrary.isAirborneFlightPhase('TAXI'));
+});
+
+ava('.isAirborneFlightPhase() returns true when passed `WAITING`', (t) => {
```
Comment (erikquinn) [inline test/navigationLibrary/NavigationLibrary.spec.js]: Same thing; you meant the reverse of what you did.

**Comment by n8rzz** [comment]
Well that's embarrassing. I had renamed that method locally but must not have pushed. I'll get that pushed tonight.

**Comment by n8rzz** [comment]
@erikquinn fixed the glitch, should read a little better now 🤕

**Comment by n8rzz** [comment]
@ij1 those are two good points. 

- I'm not a fan of that name either
- the commander changes where missed when those property names were changed in the fms, causing the `sid` and `star` commands to fail.  Oops.

**Pull request merged**